### PR TITLE
Add firewall for mac via packet filter

### PR DESCRIPTION
**Pull request by rikardev**
This is my attempt at firewall for macos with pf. Would love it if someone could take a look and verify that it works on other machines than mine. I don't write much python at my day job so any feedback on code style is very appreciated as well. Cheers.

### CODE REVIEW

File: vpn_slice/mac.py [hunk 1]
```diff
@@ -131,3 +131,91 @@ def deconfigure_domain_vpn_dns(self, domains, nameservers):
...
+            if match:
+                enable_token = match.group(1)
+
+        if not enable_token:
+            print("WARNING: failed to get pf enable reference token, packet filter might not shutdown correctly")
+
+        anchor = '{}/{}'.format(self._PF_ANCHOR, device)
+        # add anchor to generate rules with
+        with open(self._PF_CONF_FILE, 'a') as file:
+            file.write('anchor "{}" # vpn-slice-{} AUTOCREATED {}\n'.format(anchor, device, enable_token))
```
Comment (dlenski) [inline vpn_slice/mac.py:173]: Much appreciated that you're using the same comment syntax as for `/etc/hosts` :+1: :+1:

**Comment by dlenski** [comment]
> This is my attempt at firewall for macos with pf.

The code looks well-thought-out and carefully written.

My understanding is that [`pf` exists on MacOS 10.6+](https://en.wikipedia.org/wiki/PF_(firewall)#Ports). Is that correct ("supports MacOS 10.6 and newer") for this code?

> Would love it if someone could take a look and verify that it works on other machines than mine.

I don't use Mac myself, so will wait for other users to verify this. @gmacon may be interested.

**Comment by rikardev** [comment]
> My understanding is that [`pf` exists on MacOS 10.6+](https://en.wikipedia.org/wiki/PF_(firewall)#Ports). Is that correct ("supports MacOS 10.6 and newer") for this code?

Aha, yes, you're right. I'll add a check and only use this if >= 10.6.

**Comment by gmacon** [comment]
Unfortunately, I don't use vpn-slice at work any more; policies were changed, and doing so is now prohibited. It occurs to me that at least *some* parts of the macOS support should also result in support for FreeBSD, which might be a way to more easily test these components.

**Comment by dlenski** [comment]
Haven't gotten any feedback on this from other Mac users, but it looks pretty good to me, so let's merge.

**Pull request merged**